multiqc/modules/mosdepth/mosdepth.py

```diff
@@ -286,6 +337,65 @@ def parse_cov_dist(self):
                 if s_name in cumcov_dist_data:
                     self.add_data_source(f, s_name=s_name, section="genome_results")
 
+        # Applying the contig coverage cutoff. First, count the total coverage for
+        # every contig.
+        total_cov_per_contig = defaultdict(lambda: 0)
+        total_cov = 0
+        for s_name, perchrom in perchrom_avg_data.items():
+            for contig, cov in perchrom.items():
+                total_cov_per_contig[contig] += cov
+                total_cov += cov
+
+        # Now, collecting the contigs that passed the cutoff.
+        req_cov = float(total_cov) * self.cfg["perchrom_fraction_cutoff"]
+        passing_contigs = set()
+        for s_name, perchrom in perchrom_avg_data.items():
+            for contig, cov in perchrom.items():
+                if float(total_cov_per_contig[contig]) > req_cov:
+                    if contig not in passing_contigs:
+                        passing_contigs.add(contig)
+
+        rejected_contigs = set()
+        filtered_perchrom_avg_data = defaultdict(dict)
+        for s_name, perchrom in perchrom_avg_data.items():
+            for contig, cov in perchrom.items():
```

Comment:
Feels a little ugly looping over this so often - I assume it's quick though? (Thinking of that other issue where the user was running 12k mosdepth reports or something)

Reply:
Ran profiling to make sure. Those lines take a negligent amount of time compared to other places: for 100 copies of mosdepth test data, these loop run in ~1000ms each, whereas the line `if any(fnmatch.fnmatch(contig, str(pattern)) for pattern in self.cfg["exclude_contigs"])` takes 1104668ms :) Because it's run for each line in mosdepth output (`for line in f["f"].split("\n")`), and repeated millions of times because of that.

Reply:
That `fnmatch` sounds like a smoking gun! Are you addressing that somewhere?